compiler/rustc_codegen_ssa/src/debuginfo/type_names.rs

```diff
@@ -33,12 +34,13 @@ pub fn compute_debuginfo_type_name<'tcx>(
     tcx: TyCtxt<'tcx>,
     t: Ty<'tcx>,
     qualified: bool,
+    type_name_cache: &RefCell<FxHashMap<(Ty<'tcx>, bool), String>>,
```

Comment:
Can you try to make this `&mut FxHashMap<(Ty<'tcx>, bool), String>` instead of `&RefCell<...>` everywhere in this file and then do 
```rust
compute_debuginfo_type_name(cx.tcx, ty, qualified, debug_context(cx).type_name_cache.borrow_mut())
```
in `check_type_name_cache()`? It should be sufficient to secure mutable access to the cache once in the beginning.

Reply:
So I tried this, but `borrow_mut()` returns a `MappedWriteGuard`. This works if we use `&mut `debug_context(cx)..` though.

Reply:
Yes, that's what I meant. I wasn't sure if the compiler would do that automatically or if the explicit &mut would be needed.